js/foam/ui/TableView.js

```diff
@@ -647,12 +648,18 @@ CLASS({
           self.mouseOverRow = false;
           self.selection = '';
         };
-        e.onclick = function(evt) {
-          self.hardSelection = self.selection = obj;
-          self.publish(self.CLICK, obj);
+        e.onclick = function() {
+          if (timer) window.clearTimeout(timer);
+          timer = window.setTimeout(function() {
+            console.log('onclick');
+            self.hardSelection = self.selection = obj;
+            self.publish(self.CLICK, obj);
```

Comment:
One problem with this approach is that javascript prevents you from doing certain things unless you're in a click handler, such a opening a window.  If you add a timeout and someone is listening for tableview click events and trying to open popups for the rows they click on, it will no longer work.

Can you do the timer work in your client code which is listening for table view click/doubleclick events?